apps/go-mod-validator/src/github.ts

```diff
@@ -0,0 +1,114 @@
+import { Octokit } from "octokit";
+
+async function getDefaultBranch(
+  owner: string,
+  repo: string,
+  octokitClient: Octokit,
+): Promise<string> {
+  const repoObject = await octokitClient.request(
+    `GET /repos/${owner}/${repo}/`,
+  );
+  return repoObject.data.default_branch;
+}
+
+async function findCommitInDefaultBranch(
+  owner: string,
+  repo: string,
+  defaultBranch: string,
+  commitSha: string,
+  octokitClient: Octokit,
+) {
+  const compareResult = await octokitClient.request(
+    `GET /repos/${owner}/${repo}/compare/${defaultBranch}...${commitSha}`,
+  );
+  return (
+    compareResult.data.status === "identical" ||
+    compareResult.data.status === "behind"
+  );
+}
+
+async function findTagInDefaultBranch(
+  owner: string,
+  repo: string,
+  defaultBranch: string,
+  tag: string,
+  octokitClient: Octokit,
+) {
+  const repoTags = await octokitClient.request(
+    `GET /repos/${owner}/${repo}/tags`,
+  );
+  for (const repoTag of repoTags.data) {
+    if (repoTag.name != tag) {
+      continue;
+    }
+
+    // check commit is on the default branch
+    return await findCommitInDefaultBranch(
+      owner,
+      repo,
+      defaultBranch,
+      repoTag.commit.sha,
+      octokitClient,
+    );
+  }
+
+  return false;
+}
+
+async function getVersionType(versionString: string) {
+  // matches pseudo versins like v0.0.5-0.20220116011046-fa5810519dcb
```

Comment:
```suggestion
  // matches pseudo versions like v0.0.5-0.20220116011046-fa5810519dcb
```

Reply:
what makes this a pseudo version? afaik its still valid semver, prerelease format